### PR TITLE
SEO Basics

### PR DESCRIPTION
**Pull request by jpaddison3**
3 different changes:

* Markdown is the only editor able to add http images once ckEditor ships. That editor will now show a warning if you attempt to add one.
* We migrate the html of all old posts who's images can be straightforwardly upgraded to https
* We add a noIndex field to posts and a migration that sets it for all old posts with low karma.

┆Issue is synchronized with this [Trello card](https://trello.com/c/lKVZUb4Z) by [Unito](https://www.unito.io/learn-more)

### CODE REVIEW

**Comment by jimrandomh** [comment]
Before applying the noindex migration to LW, we should take a look at the stuff we imported from the Agent-Foundations forum and from SSC. A lot of the Agent-Foundations stuff has no votes at all (not even self-votes), but is high quality; those posts, we should probably make their authors upvote them like they would have if they had posted directly, and maybe have a mod go through and update any remaining good ones.  There are also some SSC posts that we imported, which only have a few votes because almost all of the attention they got was on the SSC version; for example https://www.lesswrong.com/posts/K9aLcuxAPyf5jGyFX/teachers-much-more-than-you-wanted-to-know , which I just upvoted from 3 to 5.

**Comment by jimrandomh** [comment]
(Comments I'm leaving on this PR are mostly things that are the LW team's responsibility, not the EA forum team's responsibility, though you might be interested in some of them.)

File: packages/lesswrong/server/migrations/2020-05-13-noIndexLowKarma.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { registerMigration, forEachDocumentBatchInCollection } from './migrationUtils';
+import Posts from '../../lib/collections/posts/collection';
+import moment from 'moment'
+import { forumTypeSetting } from '../../lib/instanceSettings';
+
+export const LOW_KARMA_THRESHOLD = 5
+
+const dateFormat = 'YYYY-MM-DD'
+const launchDateByForum = {
+  LessWrong: 'TODO', // lw-look-here
```
Comment (jimrandomh) [inline packages/lesswrong/server/migrations/2020-05-13-noIndexLowKarma.ts:10]: 2017-something-something

Comment (jpaddison3) [reply packages/lesswrong/server/migrations/2020-05-13-noIndexLowKarma.ts:10]: Wow, were reasonable posts getting <5 karma on old-lw?

Comment (jimrandomh) [reply packages/lesswrong/server/migrations/2020-05-13-noIndexLowKarma.ts:10]: No. But there are posts which were imported from other places and back-dated to when they were posted in those places, so they never appeared in the latest posts list at all.

**Comment by jimrandomh** [comment]
Since the https-URL enforcement is client-side, GreaterWrong users may still wind up submitting HTTP links.

**Comment by jpaddison3** [comment]
> There are also some SSC posts that we imported, which only have a few votes because almost all of the attention they got was on the SSC version

It's not obvious to me you want google indexing these. It's not obvious, but it seems like google has a sense of your site's "average quality". These posts are not linked to anywhere, which makes them (a) drag down your average, (b) extremely unlikely to show up in search results and therefore not worth indexing.

**Comment by Raemon** [comment]
Can someone clarify who's court this is in? A lot of discussion happened here but I'm not quite sure what the takeaway is.

**Comment by jpaddison3** [comment]
Pretty confident this isn't in my court

**Comment by jpaddison3** [comment]
There's a new, pretty simple merge conflict. LMK if you want me to fix it.

**Comment by Raemon** [comment]
@jimrandomh I marked this as assigned to you because you had commented a bunch so I assumed had some context, but if you prefer to assign it to me I'm up for it.

**Comment by jpaddison3** [comment]
The other obvious choice would be Oli, who has some SEO context.

**Comment by jpaddison3** [comment]
@Raemon, you're thinking about github process. Is this a case where I should ping someone?

**Comment by Raemon** [comment]
Assigning to habryka. I think Jim may be doing some deep work stuff and might take a bit longer to get around to reviewing.

**Comment by Discordius** [comment]
My current model is that we should merge this, but not run the `noIndex` migration until we saw whether it had any effect for the EA Forum or so in a month or so. We might still do it anyways afterwards, but given that we have a natural experiment, seems good to hedge a bit.

**Comment by Raemon** [comment]
Okay, so @Discordius should @jpaddison3 then fix the merge conflicts and then we approve/merge? and set yourself a reminder to check in a month about the migration?

**Comment by Discordius** [comment]
We should also run the http->https conversion script, which is great and I am super grateful for JP for making that happen. 

I will review this tomorrow. Though @jpaddison3 will have to fix the merge conflicts, since I don't have access to the branch this is descended from.

**Comment by jpaddison3** [comment]
Will do. Also y'all have write access now.

**Comment by jimrandomh** [comment]
One other issue, which is that the migration which converts images in existing posts to https is editing a denormalized field, without editing the underlying revisions collection, which will have some weird effects.

**Comment by jimrandomh** [comment]
I was treating the assignment as representing followup development work, not reviewing. If we're going to take just the subset that's ready to merge without doing any dev on it, that's fine.

**Comment by jpaddison3** [comment]
That was a conscious decision, as noted in the file docstring. I was operating under the belief that what would happen is that if someone edited the post, the post would be updated to include the insecure image again. Which, while not ideal, should be pretty rare given that most of the in-question posts are old.

**Pull request merged**